mmdet/utils/misc.py

```diff
@@ -36,3 +40,37 @@ def find_latest_checkpoint(path, suffix='pth'):
             latest = count
             latest_path = checkpoint
     return latest_path
+
+
+def update_data_root(cfg, logger=None):
+    """Update data root according to env MMDET_DATASETS.
+
+    If set env MMDET_DATASETS, update cfg.data_root according to
+    MMDET_DATASETS. Otherwise, using cfg.data_root as default.
+
+    Args:
+        cfg (mmcv.Config): The model config need to modify
+        logger (logging.Logger | str | None): the way to print msg
+    """
+    assert isinstance(cfg, mmcv.Config), \
```

Comment:
It's my mistake, please fix it as follows

```python
assert isinstance(cfg, mmcv.Config), \ ...

if 'MMDET_DATASETS'  not in os.environ:
    return

dst_root = os.environ['MMDET_DATASETS']
print_log(f'MMDET_DATASETS has been set to be {dst_root}.'
        f'Using {dst_root} as data root.')

def update(cfg, src_str, dst_str):
...
```

Reply:
@hhaAndroid
Thank you a lot !
Sorry not to review the code carefully and make this stupid mistake, I'll check it more carefully before commit next time.